### DISCUSSION TITLE
Use mimalloc as the default allocator

### DISCUSSION THREAD

**Discussion by FilipAndersson245** [typst/typst#1695] [Ideas]
Hello, I did some testing if changing the default allocator. `Mimalloc` is a microsoft developed allocator that promises to very fast.
I tested it against the default one on linux doing 10000 runs compiling the IEEE template from the website. Doing that over ~20 minutes this was the end result

![image](https://github.com/typst/typst/assets/17986183/b0397bb1-4dac-4651-862d-9d28210ab1d4)
chaining to mimalloc seems to yield a reduced runtime (6%), being a bit faster. the change to the code is minimal and the overhead of adding the dependency is quite minimal. 

What do we think?

**Comment by laurmaedje** [comment]
Sorry for only answering after you opened the PR. Copying my answer from the PR:

> The `mimalloc` library requires a C compiler to be installed. Right now Typst is pure Rust and I would prefer keeping it that way as it makes the build process simpler, faster and more platform-independent.

Still, thanks for the work you put into this!

**Reply by FilipAndersson245** [reply]
Seems fair, otherwise maybe it could be a feature to be turned on by end user, as it seem to perform better then Linux default allocator on typst code, I know from other projects it seem to perform even better compared to the windows one.
